File: Python/programmers/etc/level1/기사단원의_무기.py
def solution(number, limit, power):
    answer = [1]
    for i in range(2, number+1):
        cnt = 0
        for j in range(1, int(i**0.5)+1):
            if i%j == 0:
                cnt += 1
                if j**2 != i:
                    cnt += 1
        if cnt > limit:
            cnt = power
            answer.append(cnt)
        else:    
            answer.append(cnt)
    return sum(answer)


number, limit, power = 5, 3, 2
print(solution(number, limit, power)) # 10

number, limit, power = 10, 3, 2
print(solution(number, limit, power)) # 21

File: Java/programmers/etc/level1/기사단원의무기.java
package CodeTest.Java.programmers.etc.level1;

public class 기사단원의무기 {
    public int solution(int number, int limit, int power) {
        int answer = 1;
        
        for (int i=2; i<=number; i++) {
            int cnt = 0;
            for (int j=1; j<=Math.sqrt(i); j++) {
                if (i%j == 0) {
                    cnt++;
                    if (j*j != i) {
                        cnt++;
                    }
                }
            }
            if (cnt > limit) cnt = power;
            answer += cnt;
        }
        
        return answer;
    }
}
